src/emscripten-optimizer/simple_ast.h

```diff
@@ -396,19 +396,19 @@ struct Value {
         }
         os << '[';
         if (pretty) {
-          os << std::endl;
```

Comment:
Apparently `'\n'` gets expanded to `"\r\n"` on windows regardless, and `endl` is defind as being identical to `os << '\n' << std::flush`.

From the C++14 draft spec here (http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf , 12MB PDF):
```
namespace std {
  template <class charT, class traits>
    basic_ostream<charT,traits>& endl(basic_ostream<charT,traits>& os);
  }
}
§ 27.7.3.8 1063
1 Effects: Calls os.put(os.widen(’\n’)), then os.flush().
2 Returns: os.
```

And I cannot find any reference in the spec itself that explains why `os << '\n'` gets correctly expanded. But, from http://en.cppreference.com/w/cpp/io/basic_ostream/operator_ltlt2:
```
Before insertion, first, all characters are widened using os.widen()
```

So, TIL some things.